language/move-core/types/src/vm_status.rs

```diff
@@ -152,33 +188,28 @@ pub const EASSERT_ERROR: u64 = 42; // catch-all error code for assert failures
 // FUTURE: At the moment we can't pass transaction metadata or the signed transaction due to
 // restrictions in the two places that this function is called. We therefore just pass through what
 // we need at the moment---the sender address---but we may want/need to pass more data later on.
-pub fn convert_prologue_runtime_error(err: VMStatus, txn_sender: &AccountAddress) -> VMStatus {
-    if err.major_status == StatusCode::ABORTED {
-        let mut message = None;
-        let new_major_status = match err.sub_status {
-            Some(EACCOUNT_FROZEN) => StatusCode::SENDING_ACCOUNT_FROZEN,
-            // Invalid authentication key
-            Some(EBAD_ACCOUNT_AUTHENTICATION_KEY) => StatusCode::INVALID_AUTH_KEY,
-            // Sequence number too old
-            Some(ESEQUENCE_NUMBER_TOO_OLD) => StatusCode::SEQUENCE_NUMBER_TOO_OLD,
-            // Sequence number too new
-            Some(ESEQUENCE_NUMBER_TOO_NEW) => StatusCode::SEQUENCE_NUMBER_TOO_NEW,
-            // Sequence number too new
-            Some(EACCOUNT_DOES_NOT_EXIST) => {
-                message = Some(format!("sender address: {}", txn_sender));
-                StatusCode::SENDING_ACCOUNT_DOES_NOT_EXIST
-            }
-            // Can't pay for transaction gas deposit/fee
-            Some(ECANT_PAY_GAS_DEPOSIT) => StatusCode::INSUFFICIENT_BALANCE_FOR_TRANSACTION_FEE,
-            Some(ETRANSACTION_EXPIRED) => StatusCode::TRANSACTION_EXPIRED,
-
-            sub_status => {
-                return VMStatus::new(err.major_status, sub_status, err.message);
-            }
-        };
-        VMStatus::new(new_major_status, None, message)
-    } else {
-        err
+pub fn convert_prologue_runtime_error(status: VMStatus) -> VMStatus {
+    match status {
+        // TODO look at location? or remove this function entirely
```

Comment:
I don't know, this looks reasonable to me. If those are the only prologue errors that is great.
It also seems they don't need location or anything, just the code should be enough.
If we are not there yet, you'd hope we'll get there at some point, no?

Reply:
My thought is you don't need this function at all. You have the location + the abort code, thats MORE granular and trustable than this current implementation. It feels like an un needed layer of complexity now that we have more information

Reply:
but I think those status code are for the client as validation errors.
I guess you are saying the the client can filter properly but it seems to me this is the client.
maybe we'll briefly chat at some point but I don't think there is much to change for now